### PR TITLE
Nv 2017 onboarding start from scratch

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?

- start from scratch onboarding tour
### Other information (Screenshots)

https://github.com/novuhq/novu/assets/29251776/3a38a021-5f9e-491b-8223-f11e1bfcf148

### CODE REVIEW

File: apps/api/src/app/user/user.controller.ts [hunk 1]
```diff
@@ -84,4 +88,22 @@ export class UsersController {
       })
     );
   }
+
+  @Put('/onboarding-tour')
+  @ApiResponse(UserOnboardingTourRequestDto)
+  @ApiOperation({
+    summary: 'Update onboarding tour count',
+  })
+  @ExternalApiAccessible()
```
Comment (scopsy) [inline apps/api/src/app/user/user.controller.ts:97]: Can we hide this API From the spec? Not sure it's needed there and can confuse people I think. wdyt?

Comment (BiswaViraj) [reply apps/api/src/app/user/user.controller.ts:97]: Agree, hiding it

File: apps/web/src/pages/templates/editor/useStartFromScratchTour.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    tooltipComponent: StartFromScratchTourTooltip,
+    locale: { last: 'Got it' },
+    offset: 0,
+  },
+];
+
+export const useStartFromScratchTour = ({ startTour }: { startTour: () => void }) => {
+  const { currentUser } = useAuthContext();
+
+  const showOnboardingTour = currentUser?.showOnBoardingTour ?? 0;
```
Comment (LetItRock) [inline apps/web/src/pages/templates/editor/useStartFromScratchTour.tsx:68]: this should be moved inside the `useEffectOnce`

**Review by LetItRock — COMMENTED** [review]
Overall looks good! ⭐ 
Just added a few suggestions and found one thing when the step settings are opened: 

https://github.com/novuhq/novu/assets/2607232/eafd9812-e88d-40d9-ba5a-5d4987ce9eae

File: apps/web/cypress/tests/start-from-scratch-tour.spec.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
+describe('Digest Playground Workflow Page', function () {
```
[the remark this answers is not in the document]
Comment (BiswaViraj) [reply apps/web/cypress/tests/start-from-scratch-tour.spec.ts:1]: 🙈

File: apps/api/src/app/user/usecases/update-on-boarding-tour/update-on-boarding-tour.usecase.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      },
+      {
+        $set: {
+          showOnBoardingTour: command.showOnBoardingTour,
+        },
+      }
+    );
+
+    const user = await this.userRepository.findById(command.userId);
+    if (!user) throw new NotFoundException('User not found');
```
[the remark this answers is not in the document]
Comment (BiswaViraj) [reply apps/api/src/app/user/usecases/update-on-boarding-tour/update-on-boarding-tour.usecase.ts:30]: Agree, copied it from another use case 😅 , let me update

**Comment by BiswaViraj** [comment]
Fixed the element not found when the sidebar settings were opened by redirecting to the basePath.

@LetItRock 

https://github.com/novuhq/novu/assets/29251776/056520b0-4ec0-4695-b094-24feca45330f

**Review by LetItRock — APPROVED** [review]
looks good! 🚀

**Pull request merged**